src/metatrain/utils/scaler.py

```diff
@@ -0,0 +1,237 @@
+from typing import Dict, List, Union
+
+import metatensor.torch
+import numpy as np
+import torch
+from metatensor.torch import TensorMap
+from metatensor.torch.atomistic import ModelOutput
+
+from .additive import remove_additive
+from .data import Dataset, DatasetInfo, TargetInfo, get_all_targets
+from .jsonschema import validate
+
+
+class Scaler(torch.nn.Module):
+    """
+    A class that scales the targets of regression problems to unit standard
+    deviation.
+
+    :param model_hypers: A dictionary of model hyperparameters. The paramater is ignored
+        and is only present to be consistent with the general model API.
+    :param dataset_info: An object containing information about the dataset, including
+        target quantities and atomic types.
+    """
+
+    outputs: Dict[str, ModelOutput]
+    scales: torch.Tensor
+
+    def __init__(self, model_hypers: Dict, dataset_info: DatasetInfo):
+        super().__init__()
+
+        # `model_hypers` should be an empty dictionary
+        validate(
+            instance=model_hypers,
+            schema={"type": "object", "additionalProperties": False},
+        )
+
+        self.dataset_info = dataset_info
+
+        self.new_targets: Dict[str, TargetInfo] = dataset_info.targets
+        self.outputs: Dict[str, ModelOutput] = {}
+        self.register_buffer("scales", torch.ones((0,), dtype=torch.float64))
```

Comment:
so this is registering a buffer with no data inside, and then later changing the shape? If so, it might be worth an explanation.

Reply:
Added comment